component/azstorage/utils.go

```diff
@@ -354,6 +354,21 @@ func storeDatalakeErrToErr(err error) uint16 {
 	return ErrNoErr
 }
 
+// Convert datalake storage error to common errors
+func storeFileErrToErr(err error) uint16 {
+	if serr, ok := err.(azfile.StorageError); ok {
```

Comment:
There is a similar method already for blobs, may be we can extend the same to support this as well.

Reply:
datalake has a separate storeDatalakeErrToErr() method; i think it'd be better to have a separate method for files too since it references azfile (and the others reference azblob and azbfs)